hibernation-setup-tool.c

```diff
@@ -835,6 +849,9 @@ static bool try_zeroing_out_with_fallocate(const char *path, off_t size)
     }
 
     if (fallocate(fd, 0, 0, size) < 0) {
+        if (unlink(path) < 0)
```

Comment:
if unlink is successful - wouldn't errno get cleared, leaving nothing to be logged at 856/858?